### PR TITLE
cli: include diff summary in commit/describe editor content

### PR DESCRIPTION
**Pull request by yuja**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: tests/test_commit_command.rs [hunk 1]
```diff
@@ -54,6 +54,25 @@ fn test_commit_with_editor() {
         std::fs::read_to_string(test_env.env_root().join("editor0")).unwrap(), @r###"
     initial
 
+    JJ: This part contains the following changes:
```
Comment (martinvonz) [inline tests/test_commit_command.rs:57]: "This part" sounds weird when we're not talking about splitting a commit. I think it's clear enough to use "This commit" for all cases (`commit`, `split`, etc.). What do you think?

Comment (martinvonz) [inline tests/test_commit_command.rs:57]: We should probably special-case the empty diff and either skip this output or say something like this:

```
JJ: This commit contains the following changes:
JJ: <none>
```

Can be done later if you prefer.

**Review by martinvonz — APPROVED** [review]
Very nicely done, and nice cleanup of the tests too.

File: tests/test_commit_command.rs (hunk 1 above)
Comment (yuja) [reply tests/test_commit_command.rs]: Sounds good, done.

Comment (yuja) [reply tests/test_commit_command.rs]: Yep, removed the diff summary if commit/describe content is empty. I think that's common in "new -> describe -> do something" workflow.

**Comment by ilyagr** [comment]
Nice, thank you!

**Pull request merged**